bare_metal_billing/s3_bucket.py

```diff
@@ -0,0 +1,33 @@
+import os
+import logging
+import functools
+
+import boto3
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+@functools.lru_cache
+def get_lease_bucket():
```

Comment:
now when the code is ready to upload the generated invoice to S3, will we create these functions just for uploading?

My advice would be to make these functions a bit more generic i.e. it should get you any bucket the lease bucket or the invoice bucket, based on the arguments passed to it.

Reply:
If the bucket name is fetched from the env vars, doesn't that already let it fetch from any bucket?